How to use elements defined with `graphql_scalar!`?

The Juniper book has a lovely example of a scalar, named `UserID`, but does not show how to use it. I copied and pasted the `UserID` example, renamed it to `Digest`, and tried to use it in an object definition. It did not work out as hoped:

```
error[E0277]: the trait bound `schema::Digest: juniper::types::base::GraphQLType<__S>` is not satisfied
  --> src/schema.rs:53:10
   |
53 | #[derive(GraphQLObject)]
   |          ^^^^^^^^^^^^^ the trait `juniper::types::base::GraphQLType<__S>` is not implemented for `schema::Digest`
   |
   = help: consider adding a `where schema::Digest: juniper::types::base::GraphQLType<__S>` bound
   = note: required because of the requirements on the impl of `juniper::types::base::GraphQLType<__S>` for `std::option::Option<schema::Digest>`
   = note: required because of the requirements on the impl of `juniper::executor::IntoResolvable<'_, __S, std::option::Option<schema::Digest>, ()>` for `std::option::Option<schema::Digest>`
```

I can't find any examples to help me know where I'm going wrong. Any guidance would be appreciated, thanks.

Speaking of examples, it seems that `Value::string()` and its friend `as_string_value()` have been deprecated, but I can't for the life of me figure out how to use the suggested alternatives, so updated examples would be super.

Admittedly, I am not a Rust expert, so these could just be "you're an idiot" issues. My apologies. Keep up the great work, I hope eventually I can make use of it.

--- a/docs/book/content/types/scalars.md
+++ b/docs/book/content/types/scalars.md
@@ -17,7 +17,7 @@ use juniper::Value;
 
 struct UserID(String);
 
-juniper::graphql_scalar!(UserID {
+juniper::graphql_scalar!(UserID where Scalar = <S> {
     description: "An opaque identifier, represented as a string"
 
     resolve(&self) -> Value {
@@ -30,10 +30,16 @@ juniper::graphql_scalar!(UserID {
         v.as_scalar_value::<String>().map(|s| UserID(s.to_owned()))
     }
 
-    from_str<'a>(value: ScalarToken<'a>) -> juniper::ParseScalarResult<'a, juniper::DefaultScalarValue> {
-        <String as juniper::ParseScalarValue>::from_str(value)
+    from_str<'a>(value: ScalarToken<'a>) -> juniper::ParseScalarResult<'a, S> {
+        <String as juniper::ParseScalarValue<S>>::from_str(value)
     }
 });
+
+#[derive(juniper::GraphQLObject)]
+struct User {
+    id: UserID,
+    username: String,
+}
 
 # fn main() {}
 ```

